### PR TITLE
#95 driver config

### LINKED ISSUE #95: Extract generic driver configuration
In a driver configuration, the following properties : 
- ``db.DRIVER.uri`` 
- ``db.DRIVER.tmp.path ``
- ``db.DRIVER.client.path``
- ``db.DRIVER.print.output``
- ``db.DRIVER.save.output``
Are redundant. They could be unified, at least in the code.

### PR DESCRIPTION
**Pull request by driccio**
Extract common driver config into DatamaintainDriverConfig.
#breakingChange

Resolves #95
Reviewers : @Lysoun

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,16 +125,25 @@ Then, you may add the dependencies to ```datamaintain-core``` and the driver mod
...
+
+### Common driver configuration
+
+| Key | Description | Default value | Mandatory? | Values examples |
+|---|---|---|---|---|
+| db.uri | URI to your db server. **Database name is mandatory.**  |  | yes | ```mongodb://localhost/my-db```<br />```mongodb://localhost:8000/my-db```<br />```mongodb://username:password@localhost/my-db```<br />```mongodb+srv://server.example.com/my-db``` <br />```mongodb://my-db,my-db2:27018/my-db``` <br /> |
 | db.trust.uri | Bypass all checks that could be done on your URI because you are very sure of it and think our checks are just liars | ```false``` | no | ```true``` or ```false``` |
-### Mongo driver configuration
+| db.mongo.print.output | If true, db output will be logged. | ```false``` | no | ```true``` or ```false``` |
+| db.mongo.save.output | If true, db output will be saved in script execution report.  | ```false``` | no | ```true``` or ```false``` |
```
Comment (Lysoun) [inline README.md]: The keys are now ```db.print.output``` and ```db.save.output```

File: README.md [hunk 2]
```diff
@@ -125,16 +125,25 @@ Then, you may add the dependencies to ```datamaintain-core``` and the driver mod
...
+
+### Specific mongodb driver configuration
+
+See common driver configuration first.
```
Comment (Lysoun) [inline README.md]: I would add a link here, referring to the common driver configuration.

File: modules/cli/src/main/kotlin/datamaintain/cli/App.kt [hunk 3]
```diff
@@ -84,9 +84,9 @@ class UpdateDb(val runner: (DatamaintainConfig) -> Unit = ::defaultUpdateDbRunne
 
     private val verbose: Boolean? by option(help = "verbose").flag()
 
-    private val mongoSaveOutput: Boolean? by option(help = "save mongo output").flag()
+    private val saveDbOutput: Boolean? by option(help = "save db output").flag()
 
-    private val mongoPrintOutput: Boolean? by option(help = "print mongo output").flag()
+    private val printDbOutput: Boolean? by option(help = "print db output").flag()
```
Comment (Lysoun) [inline modules/cli/src/main/kotlin/datamaintain/cli/App.kt]: I know it should not be in the scope of this PR but I find those helps unclear. I would put "save your scripts outputs" or something like that, to remind the user that everything logged by their script (which may be something they wanted to log and logs of the driver) is concerned here.

**Review by Lysoun — CHANGES_REQUESTED** [review]
There is still a use of ```--mongo-uri``` and of ```mongo-print-output``` in the docker sample, it should be changed.
Thank you for the tests run configurations, I'm sure it will be useful!

File: modules/cli/src/main/kotlin/datamaintain/cli/App.kt (hunk 3 above)
Comment (driccio) [reply modules/cli/src/main/kotlin/datamaintain/cli/App.kt]: Agree with you but I would write 'Print your script and db outputs'. What do you think ?

Comment (Lysoun) [reply modules/cli/src/main/kotlin/datamaintain/cli/App.kt]: That's great 👍

**Review by Lysoun — APPROVED** [review]

**Pull request merged**